deploy/sdk/src/dynamo/sdk/request_tracing.py

```
@@ -0,0 +1,392 @@
+    for attr_name in dir(cls):
+        attr = getattr(cls, attr_name)
+
+        if (
+            hasattr(attr, "__call__")
+            and hasattr(attr, "_endpoint")
+            and not attr_name.startswith("_")
+        ):
+            wrapped_method = with_request_tracing(attr)
+            setattr(cls, attr_name, wrapped_method)
+
+            logger.debug(f"Added request tracing to {cls.__name__}.{attr_name}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)